core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -833,7 +1005,7 @@ public void applyLocalResult(MergingCompletableFuture allFuture, Object rv) {
 
       @Override
       public Object transformResult(Object[] results) {
-         return Arrays.stream(results);
+         return Arrays.stream(results).filter(o -> o != LOST_PLACEHOLDER);
```

Comment:
The return value inside the interceptor stack is a `Stream`, so no position involved; if the user wants to make the mapping from keys to returned values addressable, he should add the key to the returned value. Maybe this is suboptimal but that's how Galder designed the Traversable.